lib/assert.js

```diff
@@ -299,7 +295,7 @@ function expectedException(actual, expected) {
     return false;
   }
 
-  if (Object.prototype.toString.call(expected) == '[object RegExp]') {
+  if (Object.prototype.toString.call(expected) === '[object RegExp]') {
```

Comment:
I could go either way on it. (Let me know if you feel strongly one way or the other.)